gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/management/api/JdbcReporterRepository.java

```diff
@@ -107,39 +109,47 @@ public Maybe<Reporter> findById(String id) {
-        insertSpec = addQuotedField(insertSpec, COL_DOMAIN, item.getDomain(), String.class);
-        insertSpec = addQuotedField(insertSpec, COL_ENABLED, item.isEnabled(), String.class);
-        insertSpec = addQuotedField(insertSpec, COL_TYPE, item.getType(), String.class);
-        insertSpec = addQuotedField(insertSpec, COL_NAME, item.getName(), String.class);
-        insertSpec = addQuotedField(insertSpec, COL_DATA_TYPE, item.getDataType(), String.class);
-        insertSpec = addQuotedField(insertSpec, COL_CONFIG, item.getConfiguration(), String.class);
-        insertSpec = addQuotedField(insertSpec, COL_IS_SYSTEM, item.isSystem(), String.class);
-        insertSpec = addQuotedField(insertSpec, COL_CREATED_AT, dateConverter.convertTo(item.getCreatedAt(), null), LocalDateTime.class);
-        insertSpec = addQuotedField(insertSpec, COL_UPDATED_AT, dateConverter.convertTo(item.getUpdatedAt(), null), LocalDateTime.class);
-
-        Mono<Long> action = insertSpec.fetch().rowsUpdated();
-        return monoToSingle(action.as(trx::transactional))
-                .flatMap((i) -> this.findById(item.getId()).toSingle());
+        return reporterRepository.save(toJdbcEntity(item))
+                .flatMapMaybe(r -> this.findById(r.getId()))
+                .toSingle();
+//        TransactionalOperator trx = TransactionalOperator.create(tm);
+//        return Single.fromPublisher(getTemplate().insert(toJdbcEntity(item))
+//                        .flatMap(res -> maybeToMono(this.findById(res.getId())))
+//                        .as(trx::transactional));
+
+//        DatabaseClient.GenericExecuteSpec insertSpec = getTemplate().getDatabaseClient().sql(insertStatement);
+//
+//        insertSpec = addQuotedField(insertSpec, COL_ID, item.getId(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_R, item.getDomain(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_ENABLED, item.isEnabled(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_TYPE, item.getType(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_NAME, item.getName(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_DATA_TYPE, item.getDataType(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_CONFIG, item.getConfiguration(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_IS_SYSTEM, item.isSystem(), String.class);
+//        insertSpec = addQuotedField(insertSpec, COL_CREATED_AT, dateConverter.convertTo(item.getCreatedAt(), null), LocalDateTime.class);
+//        insertSpec = addQuotedField(insertSpec, COL_UPDATED_AT, dateConverter.convertTo(item.getUpdatedAt(), null), LocalDateTime.class);
+//
+//        Mono<Long> action = insertSpec.fetch().rowsUpdated();
+//        return monoToSingle(action.as(trx::transactional))
+//                .flatMap((i) -> this.findById(item.getId()).toSingle());
```

Comment:
**TODO:** if the Tests for ReporterRepository are successful for all RDBMS backend, we can remove these commented line.